x/batch/types/msg_batch.go

```diff
@@ -0,0 +1,92 @@
+package types
+
+import (
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/axelarnetwork/utils/funcs"
+	"github.com/axelarnetwork/utils/slices"
+)
+
+var _ cdctypes.UnpackInterfacesMessage = BatchRequest{}
+
+// NewBatchRequest is the constructor for BatchRequest
+func NewBatchRequest(sender sdk.AccAddress, messages []sdk.Msg) *BatchRequest {
+	return &BatchRequest{
+		Sender:   sender,
+		Messages: slices.Map(messages, func(msg sdk.Msg) cdctypes.Any { return *funcs.Must(cdctypes.NewAnyWithValue(msg)) }),
+	}
+}
+
+// Route returns the route for this message
+func (m BatchRequest) Route() string {
+	return RouterKey
+}
+
+// Type returns the type of the message
+func (m BatchRequest) Type() string {
+	return "Batch"
+}
+
+// ValidateBasic executes a stateless message validation
+func (m BatchRequest) ValidateBasic() error {
+	if err := sdk.VerifyAddressFormat(m.Sender); err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, sdkerrors.Wrap(err, "sender").Error())
+	}
+
+	if len(m.Messages) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "empty batch")
+	}
```

Comment:
we should also disallow batches of length one, they just waste gas

Reply:
I prefer allow length 1 batch for the ease of off-chain broadcasters, as they don't need to switch on the length and build txs in different ways.